roofit/roofitcore/src/RooRealVar.cxx

```diff
@@ -151,17 +151,11 @@ RooRealVar::RooRealVar(const RooRealVar& other, const char* name) :
   }
   _fast = kTRUE ;
 
-  //cout << "RooRealVar::cctor(this = " << this << " name = " << GetName() << ", other = " << &other << ")" << endl ;
-
-  RooAbsBinning* ab ;
-  TIterator* iter = other._altNonSharedBinning.MakeIterator() ;
-  while((ab=(RooAbsBinning*)iter->Next())) {
-    RooAbsBinning* abc = ab->clone() ;
-    //cout << "cloning binning " << ab << " into " << abc << endl ;
-    _altNonSharedBinning.Add(abc) ;
+  for (const auto& item : other._altNonSharedBinning) {
+    RooAbsBinning* abc = item.second->clone();
+    _altNonSharedBinning[item.first].reset(abc);
```

Comment:
In *principle* (i.e. more of a pedantic observation) the allocation should happen as an argument to `reset`:
```suggestion
    RooAbsBinning* abc = nullptr;
    _altNonSharedBinning[item.first].reset((abc =  item.second->clone()));
```
That way, exceptions are not causing any leakage. But yes, that's purely theoretical for ROOT, as long as `clone` doesn't throw.